### PR TITLE
Batch push_setting

### PR DESCRIPTION
**Pull request by ncteisen**

### CODE REVIEW

**Comment by ncteisen** [comment]
This saves on some writes by not initiating a write every time we push an updated setting out. I was going to wrap it up in my flow control optimization (#11503), but I'm seeing improvements and no regressions, so looks like its good enough to be reviewed on its own.

File: src/core/ext/transport/chttp2/transport/chttp2_transport.c [hunk 1]
```diff
@@ -518,7 +520,7 @@ static void init_transport(grpc_exec_ctx *exec_ctx, grpc_chttp2_transport *t,
                   &channel_args->args[i], settings_map[j].integer_options);
               if (value >= 0) {
                 push_setting(exec_ctx, t, settings_map[j].setting_id,
-                             (uint32_t)value);
+                             (uint32_t)value, true);
```
Comment (markdroth) [inline src/core/ext/transport/chttp2/transport/chttp2_transport.c]: Is it necessary to set `force_write` to true here, given that we unconditionally call `grpc_chttp2_initiate_write()` on line 554 below?

Comment (ncteisen) [inline src/core/ext/transport/chttp2/transport/chttp2_transport.c]: Good call! That write can actually catch all of the settings pushes from init_transport. I changed all the pushes in this function to false.

Comment (markdroth) [inline src/core/ext/transport/chttp2/transport/chttp2_transport.c]: It looks like we are now setting this parameter to false in all callers.  Given that, why not remove the parameter?  I think this entire PR could boil down to a 1-line change of simply removing the call to `grpc_chttp2_initiate_write()` from inside of `push_settings()`.

Comment (ncteisen) [inline src/core/ext/transport/chttp2/transport/chttp2_transport.c]: Done. I updated the name of the function for clarity

**Comment by ncteisen** [comment]
Mac failures: #11223

**Pull request merged**